src/librustc/infer/mod.rs

```diff
@@ -1455,7 +1455,7 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
         // Even if the type may have no inference variables, during
         // type-checking closure types are in local tables only.
         if !self.in_progress_tables.is_some() || !ty.has_closure_types() {
-            if let Some((param_env, ty)) = self.tcx.lift_to_global(&(param_env, ty)) {
+            if !(param_env, ty).has_local_value() {
```

Comment:
cc @rust-lang/compiler I think it might be better if we stop referring to these as "local values" and instead use something more like `.has_infer_vars()`?

Reply:
`has_infer_types` already exist and is a different property.